sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/QueryPlan.scala

```diff
@@ -464,6 +464,60 @@ abstract class QueryPlan[PlanType <: QueryPlan[PlanType]]
     s"($opId) $nodeName$codegenId"
   }
 
+  def extensionsInfo(mode: String = "simple"): String = {
```

Comment:
Shall we use `enum` instead of `String` type? For me, the following code in test suite looks fragile and error-prone.
```scala
val info = df.queryExecution.executedPlan.extensionsInfo("extended")
```

Reply:
Good idea. I've moved this method out of this class into `QueryExecution` because it already has the appropriate `ExplainMode` defined (it's a trait though, not an enum, but it is already available and for the same use case).